diff --git a/src/class/bth/bth_device.c b/src/class/bth/bth_device.c
--- a/src/class/bth/bth_device.c
+++ b/src/class/bth/bth_device.c
@@ -43,8 +43,10 @@ typedef struct {
   uint8_t ep_acl_in;
   uint16_t ep_acl_in_pkt_sz;
   uint8_t ep_acl_out;
+#if CFG_TUD_BTH_ISO_ALT_COUNT > 0
   uint8_t ep_voice[2];  // Not used yet
   uint8_t ep_voice_size[2][CFG_TUD_BTH_ISO_ALT_COUNT];
+#endif
 
   // Previous amount of bytes sent when issuing ZLP
   uint32_t prev_xferred_bytes;
@@ -112,8 +114,10 @@ uint16_t btd_open(uint8_t rhport, tusb_desc_interface_t const *itf_desc, uint16
 {
   tusb_desc_endpoint_t const *desc_ep;
   uint16_t drv_len = 0;
+#if CFG_TUD_BTH_ISO_ALT_COUNT > 0
   // Size of single alternative of ISO interface
   const uint16_t iso_alt_itf_size = sizeof(tusb_desc_interface_t) + 2 * sizeof(tusb_desc_endpoint_t);
+#endif
   // Size of hci interface
   const uint16_t hci_itf_size = sizeof(tusb_desc_interface_t) + 3 * sizeof(tusb_desc_endpoint_t);
   // Ensure this is BT Primary Controller
@@ -149,13 +153,14 @@ uint16_t btd_open(uint8_t rhport, tusb_desc_interface_t const *itf_desc, uint16
     desc_ep_acl_in = (tusb_desc_endpoint_t const *)tu_desc_next(desc_ep_acl_in);
   }
 
-  itf_desc = (tusb_desc_interface_t const *)tu_desc_next(tu_desc_next(desc_ep));
-
   // Prepare for incoming data from host
   TU_ASSERT(usbd_edpt_xfer(rhport, _btd_itf.ep_acl_out, _btd_epbuf.epout_buf, CFG_TUD_BTH_DATA_EPSIZE), 0);
 
   drv_len = hci_itf_size;
 
+#if CFG_TUD_BTH_ISO_ALT_COUNT > 0
+  itf_desc = (tusb_desc_interface_t const *)tu_desc_next(tu_desc_next(desc_ep));
+
   // Ensure this is still BT Primary Controller
   TU_ASSERT(TUSB_CLASS_WIRELESS_CONTROLLER == itf_desc->bInterfaceClass &&
             TUD_BT_APP_SUBCLASS == itf_desc->bInterfaceSubClass &&
@@ -208,6 +213,7 @@ uint16_t btd_open(uint8_t rhport, tusb_desc_interface_t const *itf_desc, uint16
     _btd_itf.ep_voice_size[dir][itf_desc->bAlternateSetting] = (uint8_t) tu_edpt_packet_size(desc_ep);
     drv_len += iso_alt_itf_size;
   }
+#endif
 
   return drv_len;
 }
